benches/nexmark/main.rs

```diff
@@ -1 +1,67 @@
+//! Nexmark benchmarks for DBSP
+//!
+//! Based on the queries defined at
+#![feature(is_some_with)]
+use anyhow::Result;
+use clap::Parser;
+use dbsp::{
+    circuit::Root,
+    nexmark::{
+        config::Config as NexmarkConfig, generator::config::Config as GeneratorConfig,
+        model::Event, queries::q1, NexmarkSource,
+    },
+    profile::CPUProfiler,
+    trace::{ord::OrdZSet, BatchReader},
+};
+use rand::prelude::ThreadRng;
 
+// TODO: Enable running specific tests
+fn main() -> Result<()> {
+    let nexmark_config = NexmarkConfig::parse();
+    let generator_config = GeneratorConfig::new(nexmark_config, 0, 0, 0, 0);
+
+    let cpu_profiler = CPUProfiler::new();
+    let root = Root::build(|circuit| {
```

Comment:
We should wrap this in an instance of `Runtime`, to enable multithreading.

Reply:
OK, I'll have a look at `Runtime` today. Not yet clear to me how a continuous source like the Nexmark one can be run in different threads, but maybe that's related to the "feeding data from external sources" pain-point comment.

Reply:
We'd have to setup the circuit to run the generator in worker thread 0 only. But yeah, the "feeding data from external sources" feature will solve this properly.